src/core/cgroup.c

```diff
@@ -4062,38 +4063,48 @@ static int unit_get_memory_peak_raw(Unit *u, uint64_t *ret) {
         if (!r)
                 return -ENODATA;
 
-        return cg_get_attribute_as_uint64("memory", u->cgroup_path, "memory.peak", ret);
+        return cg_get_attribute_as_uint64("memory", u->cgroup_path, mem_attribute, ret);
 }
 
-int unit_get_memory_peak(Unit *u, uint64_t *ret) {
+static int unit_get_memory_attr_cached(Unit *u, uint64_t* last, const char* mem_attribute, uint64_t *ret) {
```

Comment:
Make mem_attribute the second parameter?

Reply:
Thanks. That's sensible.